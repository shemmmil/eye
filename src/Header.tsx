import React from 'react';
import styled from 'styled-components';
import { Button } from 'antd';
import { PlusOutlined, UserOutlined } from '@ant-design/icons';
import { Money } from './Money';

const Container = styled.div`
  width: 100%;
  color: #000;
  font-size: 15px;
  background-color: #fff;
`;
const Content = styled.div`
  height: 70px;
  display: flex;
  align-items: center;
  border-bottom: 1px solid #e5e5e5;
  padding-right: 25px;
`;
const Logo = styled.div`
  width: 100px;
  height: 70px;
  background-color: pink;
`;
const Middle = styled.div`
  width: 100%;
`;
const List = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
`;
const Item = styled.li`
  display: inline-block;

  & + li {
    margin-left: 32px;
  }
`;

const Navigation = styled.div`
  display: flex;
  align-items: center;
`;

const CreateButton = styled(Button)`
  margin-right: 32px;
`;

const User = styled.div`
  display: flex;
  align-items: center;
`;

const Avatar = styled.div`
  width: 42px;
  height: 42px;
  border-radius: 50%;
  background-color: #d9d9d9;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1.4rem;
  color: #fff;
`;

const UserName = styled.span`
  margin-left: 10px;
  &:hover {
    cursor: pointer;
  }
`;

export const Header = () => {
  return (
    <Container>
      <Content>
        <Logo />
        <Middle>
          <List>
            <Item>Мои опросы</Item>
            <Item>Помощь</Item>
            <Item>Оставить отзыв</Item>
            <Item>Блог</Item>
            <Item>Мой профиль</Item>
          </List>
        </Middle>
        <Navigation>
          <CreateButton type="primary" data-button="create-a-survey">
            Создать опрос
          </CreateButton>
          <Money />
          <User>
            <Avatar>
              <UserOutlined />
            </Avatar>
            <UserName>beta.user</UserName>
          </User>
        </Navigation>
      </Content>
    </Container>
  );
};
