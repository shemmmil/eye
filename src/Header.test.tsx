import React from 'react';
import { mount } from 'enzyme';
import { Header } from './Header';

describe('Header Render', () => {
  it('Middle navigation list items correctly renders', () => {
    const wrapper = mount(<Header />);
    expect(wrapper.find('li')).toHaveLength(5);
  });
});
