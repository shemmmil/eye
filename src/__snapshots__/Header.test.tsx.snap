// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Render Middle navigation list items correctly renders 1`] = `
.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: 6px;
  margin-right: 32px;
}

.c0 {
  width: 100%;
  color: #000;
  font-size: 15px;
  background-color: #fff;
}

.c1 {
  height: 70px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #e5e5e5;
  padding-right: 25px;
}

.c2 {
  width: 100px;
  height: 70px;
  background-color: pink;
}

.c3 {
  width: 100%;
}

.c4 {
  list-style: none;
  margin: 0;
  padding: 0;
}

.c5 {
  display: inline-block;
}

.c5 + li {
  margin-left: 32px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  margin-right: 32px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 {
  width: 42px;
  height: 42px;
  border-radius: 50%;
  background-color: #d9d9d9;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.4rem;
  color: #fff;
}

.c11 {
  margin-left: 10px;
}

.c11:hover {
  cursor: pointer;
}

<Component>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          <styled.div>
            <div
              className="c2"
            />
          </styled.div>
          <styled.div>
            <div
              className="c3"
            >
              <styled.ul>
                <ul
                  className="c4"
                >
                  <styled.li>
                    <li
                      className="c5"
                    >
                      Мои опросы
                    </li>
                  </styled.li>
                  <styled.li>
                    <li
                      className="c5"
                    >
                      Помощь
                    </li>
                  </styled.li>
                  <styled.li>
                    <li
                      className="c5"
                    >
                      Оставить отзыв
                    </li>
                  </styled.li>
                  <styled.li>
                    <li
                      className="c5"
                    >
                      Блог
                    </li>
                  </styled.li>
                  <styled.li>
                    <li
                      className="c5"
                    >
                      Мой профиль
                    </li>
                  </styled.li>
                </ul>
              </styled.ul>
            </div>
          </styled.div>
          <styled.div>
            <div
              className="c6"
            >
              <Styled(Button)
                data-button="create-a-survey"
                type="primary"
              >
                <Button
                  block={false}
                  className="c7"
                  data-button="create-a-survey"
                  ghost={false}
                  htmlType="button"
                  loading={false}
                  type="primary"
                >
                  <Wave>
                    <button
                      className="ant-btn c7 ant-btn-primary"
                      data-button="create-a-survey"
                      onClick={[Function]}
                      type="button"
                    >
                      <span
                        key=".0"
                      >
                        Создать опрос
                      </span>
                    </button>
                  </Wave>
                </Button>
              </Styled(Button)>
              <Component>
                <Button
                  block={false}
                  ghost={false}
                  htmlType="button"
                  icon={<ForwardRef(PlusOutlined) />}
                  loading={false}
                  onClick={[Function]}
                  shape="circle"
                >
                  <Wave>
                    <button
                      className="ant-btn ant-btn-circle ant-btn-icon-only"
                      onClick={[Function]}
                      type="button"
                    >
                      <ForwardRef(PlusOutlined)>
                        <AntdIcon
                          icon={
                            Object {
                              "icon": Object {
                                "attrs": Object {
                                  "focusable": "false",
                                  "viewBox": "64 64 896 896",
                                },
                                "children": Array [
                                  Object {
                                    "attrs": Object {},
                                    "children": Array [
                                      Object {
                                        "attrs": Object {},
                                        "tag": "style",
                                      },
                                    ],
                                    "tag": "defs",
                                  },
                                  Object {
                                    "attrs": Object {
                                      "d": "M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z",
                                    },
                                    "tag": "path",
                                  },
                                  Object {
                                    "attrs": Object {
                                      "d": "M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z",
                                    },
                                    "tag": "path",
                                  },
                                ],
                                "tag": "svg",
                              },
                              "name": "plus",
                              "theme": "outlined",
                            }
                          }
                        >
                          <span
                            aria-label="plus"
                            className="anticon anticon-plus"
                            role="img"
                          >
                            <IconReact
                              className=""
                              icon={
                                Object {
                                  "icon": Object {
                                    "attrs": Object {
                                      "focusable": "false",
                                      "viewBox": "64 64 896 896",
                                    },
                                    "children": Array [
                                      Object {
                                        "attrs": Object {},
                                        "children": Array [
                                          Object {
                                            "attrs": Object {},
                                            "tag": "style",
                                          },
                                        ],
                                        "tag": "defs",
                                      },
                                      Object {
                                        "attrs": Object {
                                          "d": "M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z",
                                        },
                                        "tag": "path",
                                      },
                                      Object {
                                        "attrs": Object {
                                          "d": "M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z",
                                        },
                                        "tag": "path",
                                      },
                                    ],
                                    "tag": "svg",
                                  },
                                  "name": "plus",
                                  "theme": "outlined",
                                }
                              }
                            >
                              <svg
                                aria-hidden="true"
                                className=""
                                data-icon="plus"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                key="svg-plus"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <defs
                                  key="svg-plus-svg-0"
                                >
                                  <style
                                    key="svg-plus-svg-0-defs-0"
                                  />
                                </defs>
                                <path
                                  d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                                  key="svg-plus-svg-1"
                                />
                                <path
                                  d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                                  key="svg-plus-svg-2"
                                />
                              </svg>
                            </IconReact>
                          </span>
                        </AntdIcon>
                      </ForwardRef(PlusOutlined)>
                    </button>
                  </Wave>
                </Button>
                <styled.span>
                  <span
                    className="c8"
                  >
                    0 ₽
                  </span>
                </styled.span>
                <Modal
                  confirmLoading={false}
                  maskTransitionName="fade"
                  okType="primary"
                  onCancel={[Function]}
                  title="Остаток на счете: 0 ₽"
                  transitionName="zoom"
                  visible={false}
                  width={520}
                >
                  <Component
                    closeIcon={
                      <span
                        className="ant-modal-close-x"
                      >
                        <ForwardRef(CloseOutlined)
                          className="ant-modal-close-icon"
                        />
                      </span>
                    }
                    confirmLoading={false}
                    footer={
                      <LocaleReceiver
                        componentName="Modal"
                        defaultLocale={
                          Object {
                            "cancelText": "Cancel",
                            "justOkText": "OK",
                            "okText": "OK",
                          }
                        }
                      >
                        [Function]
                      </LocaleReceiver>
                    }
                    maskTransitionName="fade"
                    okType="primary"
                    onCancel={[Function]}
                    onClose={[Function]}
                    prefixCls="ant-modal"
                    title="Остаток на счете: 0 ₽"
                    transitionName="zoom"
                    visible={false}
                    width={520}
                    wrapClassName=""
                  >
                    <PortalWrapper
                      visible={false}
                    />
                  </Component>
                </Modal>
              </Component>
              <styled.div>
                <div
                  className="c9"
                >
                  <styled.div>
                    <div
                      className="c10"
                    >
                      <ForwardRef(UserOutlined)>
                        <AntdIcon
                          icon={
                            Object {
                              "icon": Object {
                                "attrs": Object {
                                  "focusable": "false",
                                  "viewBox": "64 64 896 896",
                                },
                                "children": Array [
                                  Object {
                                    "attrs": Object {
                                      "d": "M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z",
                                    },
                                    "tag": "path",
                                  },
                                ],
                                "tag": "svg",
                              },
                              "name": "user",
                              "theme": "outlined",
                            }
                          }
                        >
                          <span
                            aria-label="user"
                            className="anticon anticon-user"
                            role="img"
                          >
                            <IconReact
                              className=""
                              icon={
                                Object {
                                  "icon": Object {
                                    "attrs": Object {
                                      "focusable": "false",
                                      "viewBox": "64 64 896 896",
                                    },
                                    "children": Array [
                                      Object {
                                        "attrs": Object {
                                          "d": "M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z",
                                        },
                                        "tag": "path",
                                      },
                                    ],
                                    "tag": "svg",
                                  },
                                  "name": "user",
                                  "theme": "outlined",
                                }
                              }
                            >
                              <svg
                                aria-hidden="true"
                                className=""
                                data-icon="user"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                key="svg-user"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                                  key="svg-user-svg-0"
                                />
                              </svg>
                            </IconReact>
                          </span>
                        </AntdIcon>
                      </ForwardRef(UserOutlined)>
                    </div>
                  </styled.div>
                  <styled.span>
                    <span
                      className="c11"
                    >
                      beta.user
                    </span>
                  </styled.span>
                </div>
              </styled.div>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Component>
`;
